cli/tests/unit/fetch_test.ts

```diff
@@ -956,3 +956,39 @@ unitTest(
     client.close();
   },
 );
+
+unitTest(
+  { perms: { net: true, read: true } },
+  async function fetchCustomHttpClientTimeoutError(): Promise<
+    void
+  > {
+    const client = Deno.createHttpClient(
+      { timeout: 1 },
+    );
+    await assertThrowsAsync(async (): Promise<void> => {
+      const _response = await fetch(
+        "https://localhost:4545/cli/tests/fixture.json",
+        { client },
+      );
+    });
+    client.close();
+  },
```

Comment:
Looks good, what's the error class that is thrown in case of timeout?

Reply:
One thing I worry about is that this test might be flaky - maybe we could add special endpoint to `test_server` that sleeps for a few ms, to ensure that it works break randomly?

Reply:
It's a `Deno.errors.Http`, the message is `error sending request for url (https://localhost:4545/cli/tests/fixture.json): operation timed out`. I haven't completely wrapped my head around building other error classes but I'm happy to change it.

Can you point me to the test_server with the sleep? I think that's a great idea I just couldn't find it. ;)

Reply:
Actually, that's ok, can you just add assertion on the error class to `assertThrowsAsync`?

Reply:
@bartlomieju How about a "sleep" param on the echo test server?

Reply:
The Timeout test checks for `Deno.errors.Http` now.